### ISSUE TITLE
question: which part of your code takes care of proxy_lr vs base_lr?

### ISSUE THREAD

**Issue by monajalal** [euwern/proxynca_pp#13]
Hello, 
In your paper you tune proxy_lr also as a hyperparameter. I was unsure where in your code you define proxy_lr could you please guide me?
https://github.com/euwern/proxynca_pp/blob/master/train.py

**Comment by euwern** [comment]
See lines 307 (base_lr) and 311 (proxy_lr). These values are set in the config file (e.g.: cub.json - lines 43 (proxy_lr) and line 46 (base_lr)).

**Comment by monajalal** [comment]
thanks for your response.

Did you mean that line 311 is the base_lr since it has the word base in it?
<img width="471" alt="Screen Shot 2021-09-23 at 9 00 52 PM" src="https://user-images.githubusercontent.com/1892917/134602854-a5ea1362-b1ea-4399-80c2-75639865e825.png">

<img width="300" alt="Screen Shot 2021-09-23 at 9 01 18 PM" src="https://user-images.githubusercontent.com/1892917/134602870-72d6a6c2-ab7f-4796-b1f3-ab8794827f75.png">

and also,  I am not sure where proxy_lr is used in the code. I found this but just said lr. Could you please point me to it?

<img width="345" alt="Screen Shot 2021-09-23 at 9 02 14 PM" src="https://user-images.githubusercontent.com/1892917/134602909-d6eb841c-122e-4658-839c-1607538dad64.png">

**Comment by euwern** [comment]
It is used by the optimizer (train.py - line 290 ) "config['opt']['type']", which is defined in the config file (cub.json - line 33) "torch.optim.Adam".  Do checkout the Pytorch documentation regarding on how to use an optimizer. 
The proxy layer has a different learning rate than the rest of the layers. 

I hope it helps.
